### PR TITLE
Update kube proxy

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
- Replace kube-proxy manifests with the ones used by kubeadm
- Use templating to use the cluster version also for kube-proxy
- Set the workerName in the addonInstallerController

**Release note**:
```release-note
Update kube-proxy addon to match the cluster version
```

### CODE REVIEW

File: addons/kube-proxy/clusterrolebinding.yaml [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
+kind: ClusterRoleBinding
 metadata:
-  name: system:kube-proxy
-  labels:
-    addonmanager.kubernetes.io/mode: "Reconcile"
-subjects:
-  - kind: Group
-    name: system:nodes
-    namespace: kube-system
+  name: kubeadm:node-proxier
```
Comment (alvaroaleman) [inline addons/kube-proxy/clusterrolebinding.yaml:4]: Was this change intended?

File: addons/kube-proxy/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      creationTimestamp: null
+      labels:
+        k8s-app: kube-proxy
+    spec:
+      containers:
+      - command:
+        - /usr/local/bin/kube-proxy
+        - --config=/var/lib/kube-proxy/config.conf
+        image: k8s.gcr.io/kube-proxy-amd64:v{{ .Cluster.Spec.Version }}
```
Comment (alvaroaleman) [inline addons/kube-proxy/daemonset.yaml:25]: Did you test this config works with 1.9 clusters?

File: api/cmd/kubermatic-controller-manager/main.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/net"
-	kuberinformers "k8s.io/client-go/informers"
```
Comment (alvaroaleman) [inline api/cmd/kubermatic-controller-manager/main.go:31]: Nice typo :D

File: api/pkg/controller/addoninstaller/addoninstaller_controller.go [hunk 4]
```diff
@@ -134,7 +138,13 @@ func (c *Controller) createDefaultAddon(addon string, cluster *kubermaticv1.Clus
 				Kind:       "Cluster",
 			},
 		},
-	})
+	}
+
+	if c.workerName != "" {
+		a.Labels[kubermaticv1.WorkerNameLabelKey] = c.workerName
```
Comment (alvaroaleman) [inline api/pkg/controller/addoninstaller/addoninstaller_controller.go:144]: Why does the addon need that annotation?

File: addons/kube-proxy/clusterrolebinding.yaml (hunk 1 above)
Comment (mrIncompetent) [reply addons/kube-proxy/clusterrolebinding.yaml:4]: Yes. The former CRB was only needed as we used the same kubeconfig as the kubelet - also giving the kubelet more permission as needed.
With this PR, kube-proxy will use the token from the ServiceAccount.

File: api/pkg/controller/addoninstaller/addoninstaller_controller.go (hunk 4 above)
Comment (mrIncompetent) [reply api/pkg/controller/addoninstaller/addoninstaller_controller.go:144]: The lister, which gets passed into the controller is already filtered by the workerName.
Means:
The addonInstallerController wants to create a Addon, looks if it already exists (Checks against the lister -> does not exist) & creates it (which fails as it actually already exists).

File: addons/kube-proxy/clusterrolebinding.yaml (hunk 1 above)
Comment (alvaroaleman) [reply addons/kube-proxy/clusterrolebinding.yaml:4]: I meant: Did you intend to name the CRB `kubeadm:node-proxier` instead of `system:kube-proxy`?

Comment (mrIncompetent) [reply addons/kube-proxy/clusterrolebinding.yaml:4]: Oh sorry, yes. The kube-proxy as a Daemonset is completely copied from kubeadm.
I would therefore stick with the exact naming.

File: addons/kube-proxy/daemonset.yaml (hunk 2 above)
Comment (mrIncompetent) [reply addons/kube-proxy/daemonset.yaml:25]: Yes, i created a v1.9.10 cluster & deployed the guestbook example.
Everything worked & the kube-proxy logs showed no errors, nor warnings

**Review by alvaroaleman — DISMISSED** [review]

**Comment by mrIncompetent** [comment]
@alvaroaleman i updated the tag. PTAL

**Review by alvaroaleman — APPROVED** [review]
